File: 0283-move-zeroes/0283-move-zeroes.py
#283
class Solution:
    def moveZeroes(self, nums: list[int]) -> None:
        insert_pos = 0

        for i in range(len(nums)):
            if nums[i] != 0:
                nums[insert_pos] = nums[i]
                insert_pos += 1

        for i in range(insert_pos, len(nums)):
            nums[i] = 0

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int indexptr = 0;
        for ( int i = 0 ; i<nums.size() ; i++){
            if (nums[i] != 0){
                std::swap(nums[i],nums[indexptr]);
                indexptr++;
            }
        }
    }
};